### DISCUSSION TITLE
meshing urban areas

### DISCUSSION THREAD

**Discussion by danielpeter** [SPECFEM/specfem3d#1575] [General]
How could we mesh urban areas? 

In end-to-end simulations, it would be interesting to include the effect of city buildings and other urban elements on the seismic wavefield, as could be done in a hero run for an earthquake scenario case. So, how could we include:
* buildings
* critical infrastructure
* or near-surface (non-linear) soil descriptions

for now, let's just collect ideas on how to approach this, links to interesting publications, and related info.

**Comment by danielpeter** [comment]
some publication links that came out of yesterday's monthly zoom meeting:
* Krishnan et al. (2006): Performance of Two 18-Story Steel Moment-Frame Buildings in Southern California during Two Large Simulated San Andreas Earthquakes.
https://journals.sagepub.com/doi/10.1193/1.2360698

* Krishnan et al. (2006): Case Studies of Damage to Tall Steel Moment-Frame Buildings in Southern California during Large San Andreas Earthquakes.
https://pubs.geoscienceworld.org/ssa/bssa/article/96/4A/1523/146702/case-studies-of-damage-to-tall-steel-moment-frame

* Taborda & Bielak (2011): Full 3D integration of site-city effects in regional scale earthquake simulations. https://bwk.kuleuven.be/apps/bwm/eurodyn2011/papers/MS03-241.pdf

* Guidotti et al. (2012): 3D numerical simulation of the Site-City Interaction during the 22 February 2011 MW 6.2
Christchurch earthquake. 
https://www.iitk.ac.in/nicee/wcee/article/WCEE2012_2290.pdf

* Zhang et al. (2020): A computational workflow for rupture-to-structural-response simulation and its application to Istanbul. https://onlinelibrary.wiley.com/doi/pdf/10.1002/eqe.3377

**Comment by elifo** [comment]
Hi Daniel,

* Computationally the structural stiffness/dynamic behavior is approximated by mesh elements such as beams, columns, and trusses. 

* The keyword for for structural analyses is soil structure interaction (SSI). The studies I am familiar with, similar to some of the linked papers above, apply domain reduction method (DRM) of Bielak et al. https://doi.org/10.1785/0120010251 which approximate the seismic radiation as an incident field and focus on advanced structural analysis. 

  Here is a recent development (Seismo-VLab) with applications: https://seismovlab.com/gallery.html
  The related PhD dissertation: Kusanovic (2021) https://thesis.library.caltech.edu/14019/1/kusanovic_danilo_2021_thesis.pdf

* Soil nonlinearity can be prescribed as a different material type, or a rheological model, similar to my implementations in 1D and 2D spectral element codes detailed below. Both my implementations use the 3D rheological model of extended IWAN (Joyner, https://doi.org/10.1785/BSSA0650051337).
  
  2D code related commit: https://github.com/elifo/sem2dpack/commit/54e72714d43f5d586f98c9c7e8f4371046020f64
  1D code: https://github.com/elifo/1D3CSEM
  My ongoing work in 3D (specfem3d): https://github.com/elifo/3D_NL/commits/master
  2D examples and publications: https://github.com/elifo/sem2dpack/tree/master/EXAMPLES/SiteEffects
  1D examples: Oral, 2016 (Chapter 3) https://theses.hal.science/tel-01562279/document

I think that specfem3d can handle the coupling of structural analysis with urban-scale complex rupture and wave propagation without DRM. It mainly lacks the formulations of SSI and structural dynamics, in addition to structural mesh.  

Elif
